proto/internal/temporal/server/api/persistence/v1/hsm.proto

```diff
@@ -161,3 +168,39 @@ message StateMachinePath {
     repeated StateMachineKey path = 1;
 }
 
+// StateMachineOperation represents a history record of operations performed on nodes in the state machine tree
+message StateMachineOperation {
+    // OperationType defines the type of operation performed
+    enum OperationType {
+        // Unspecified operation type - should not be used
+        OPERATION_TYPE_UNSPECIFIED = 0;
+        // Node was deleted from the tree
+        OPERATION_TYPE_DELETE = 1;
+        // Node state was transitioned
+        OPERATION_TYPE_TRANSITION = 2;
+    }
+
+    // Path to the node in the tree, represented as string segments in the format "type/id"
+    repeated string path = 1;
+
+    // Type of operation performed
+    OperationType operation_type = 2;
+
+    // When the operation was performed
+    google.protobuf.Timestamp operation_time = 3;
+
+    // Version and transition information for the operation
+    VersionedTransition versioned_transition = 4;
+
+    // Serialized state data at time of operation
+    bytes data = 5;
+
+    // Child nodes at time of operation
+    map<string, StateMachineMap> children = 6;
```

Comment:
We only store and replciate the last state, no need to keep the previous snapshots.

Reply:
Removed